frappe/patches/v13_0/encrypt_2fa_secrets.py

```diff
@@ -0,0 +1,45 @@
+import frappe
+import frappe.defaults
+from frappe.cache_manager import clear_defaults_cache
+from frappe.twofactor import PARENT_FOR_DEFAULTS
+from frappe.utils.password import encrypt
+
+DOCTYPE = "DefaultValue"
+OLD_PARENT = "__default"
+
+
+def execute():
+	table = frappe.qb.DocType(DOCTYPE)
+
+	# set parent for `*_otplogin`
+	(
+		frappe.qb.update(table)
+		.set(table.parent, PARENT_FOR_DEFAULTS)
+		.where(table.parent == OLD_PARENT)
+		.where(table.defkey.like("%_otplogin"))
+	).run()
+
+	# update records for `*_otpsecret`
+	secrets = {
+		key: value
+		for key, value in frappe.defaults.get_defaults_for(parent=OLD_PARENT).items()
+		if key.endswith("_otpsecret")
+	}
+
+	if not secrets:
+		return
+
+	defvalue_cases = frappe.qb.terms.Case()
+
+	for key, value in secrets.items():
+		defvalue_cases.when(table.defkey == key, encrypt(value))
+
+	(
+		frappe.qb.update(table)
+		.set(table.parent, PARENT_FOR_DEFAULTS)
+		.set(table.defvalue, defvalue_cases)
+		.where(table.parent == OLD_PARENT)
+		.where(table.defkey.like("%_otpsecret"))
```

Comment:
is this where condition needed?

the defkeys are already considered in the case when queries 🤔

Reply:
right, will remove

Reply:
turns out this WAS required.. FML 😢 
https://github.com/frappe/frappe/issues/17571

Reply:
yep - FML as well

Reply:
https://github.com/frappe/frappe/pull/17575